digest/src/core_api/ct_variable.rs

```diff
@@ -165,3 +165,26 @@ where
         write!(f, "{}", OutSize::USIZE)
     }
 }
+
+impl<T, OutSize> SerializableState for CtVariableCoreWrapper<T, OutSize>
```

Comment:
Do not forget to also implement it for `CoreWrapper` and ideally for `RtVariableCoreWrapper`.

Reply:
`CoreWrapper` has [`decompose`](https://docs.rs/digest/0.10.3/digest/core_api/struct.CoreWrapper.html#method.decompose) method that returns buffered input and `CtVariableCoreWrapper` object, and [`from_core`](https://docs.rs/digest/0.10.3/digest/core_api/struct.CoreWrapper.html#method.from_core) method to restore the object.

Do we need additionally implement `SerializableState`? Also, we do not know the size of buffered input (or we need to store the length with the partially filled block of buffered input).

Reply:
I think, yes. It would be weird to support `SerializableState` only for the core types and force users to jump through artificial error-prone hoops.

> Also, we do not know the size of buffered input (or we need to store the length with the partially filled block of buffered input).

You should attach to serialized state number of unprocessed bytes and zero-padded unprocessed data (you could use `buffer.clone().pad_with_zeros().clone()`). For SHA-256 it would mean always attaching 65 bytes, even when there are no unprocessed bytes in it. We could save one byte for the eager hashes by using the fact that the last byte will be never used, but I think it's too small of a win for the very niche feature, so it does not worth to introduce the additional complexity.